collections.go

```diff
@@ -39,14 +39,18 @@ type Collection struct {
 		- price-asc: By price, in ascending order (lowest - highest).
 		- price-desc: By price, in descending order (highest - lowest).
 	*/
+	// Rules is the list of rules that define what products go into the smart collection.
+	Rules     Rules
```

Comment:
this exists on smart collections, not collections, I think you need to create a whole new type with repositories to do smart collections as that is a separate thing in shopify